app/views/email_authentications/invitations/new.html.erb

```diff
@@ -15,27 +15,32 @@
         <div id="facility-access-power-user" class="hidden">
           Admin has access to all facilities in the system
         </div>
-        <div id="facility-access">
+        <div id="facility-access" data-page="new">
           <div class="card access-tree">
-            <div class="select-all-access-items">
-              <%= form.check_box :select_all_facilities, id: "select-all-facilities" %>
+            <div id="select-all-facilities" hidden="hidden">
+              <%= form.check_box :select_all_facilities, id: "select-all-facilities-input" %>
             </div>
 
-            <% access_tree = current_admin.visible_access_tree %>
-            <%= render access_tree[:render_partial],
-                tree: access_tree[:data],
-                user_being_edited: nil,
-                tree_depth: 0,
-                root: access_tree[:root],
-                page: :new,
-                form: form %>
+            <%= render_async(access_tree_admin_path(current_admin, page: :new),
+                error_message: '<p class="load-error-message">There was an error, <a href=""> please try again </a></p>') do %>
+              <div class="spinner-wrapper">
+                <div class="spinner spinner-border text-primary" role="status">
+                  <span class="sr-only">Loading...</span>
+                </div>
+              </div>
+            <% end %>
           </div>
         </div>
 
         <%= form.submit "Send Invite", class: "btn btn-primary", id: "form-submit", hidden: :hidden %>
       <% end %>
     </div>
   </div>
+  <script>
+    const adminAccessInvite = new AdminAccessInvite()
```

Comment:
Similar to above, I'd love to pass handles to the necessary DOM elements as constructor params. What do you think?